oxide-auth/src/primitives/generator.rs

```diff
@@ -132,24 +132,24 @@ impl Assertion {
     /// padding or shortening of the supplied key material may be applied in the form dictated by
     /// the signature type. See the respective standards.
     ///
-    /// If future suites are added where this is not possible, his function may panic when supplied
+    /// If future suites are added where this is not possible, this function may panic when supplied
     /// with an incorrect key length.
+    /// 
+    /// Currently, the implementation lacks the ability to really make use of another hasing mechanism than
+    /// hmac + sha256.
     pub fn new(kind: AssertionKind, key: &[u8]) -> Self {
-        let key = match kind {
-            AssertionKind::HmacSha256 => SigningKey::new(&SHA256, key),
+        match kind {
+            AssertionKind::HmacSha256 => Assertion { key: key.to_vec() },
```

Comment:
`SigningKey::new` did perform some essential verification and transformation depending on the length, where has this gone? Read [the documentation of `ring`](https://docs.rs/ring/0.16.12/ring/hmac/struct.Key.html#method.new) on what it does and which standard this implements.

Reply:
https://github.com/RustCrypto/MACs/blob/master/hmac/src/lib.rs#L142
```
        // The key that Hmac processes must be the same as the block size of the
        // underlying Digest. If the provided key is smaller than that, we just
        // pad it with zeros. If its larger, we hash it and then pad it with
        // zeros.
```
Hmac seems to pad or truncate (via hashing) the key on its own. I did not think it was necessary to actually do it with the key. Am I missing something?